ci/test_codeql.py

```diff
@@ -0,0 +1,190 @@
+"""
+
+import os
+import subprocess
+import sys
+from pathlib import Path
+import json
+import yaml
+from termcolor import colored
+
+
+class CodeQLAnalyzer:
+    """Encapsulates the logic for running CodeQL analysis."""
+
+    def __init__(self, workdir):
+        """Initializes the analyzer with the working directory."""
+        if not workdir:
+            raise ValueError("WORKDIR environment variable is not set.")
+        self.repo_path = Path(workdir).resolve()
+        if not self.repo_path.is_dir():
+            raise ValueError(f"WORKDIR '{self.repo_path}' is not a valid directory.")
+        self.output_path = self.repo_path / "dist" / "codeql"
+        if not self.output_path.is_dir():
+            os.makedirs(self.output_path)
+
+    def _print_detailed_summary(self, results_path):
+        """Prints a detailed summary of the issues found in a SARIF file."""
+        try:
+            with open(results_path, "r", encoding="utf-8") as f:
+                data = json.load(f)
+
+            if not data["runs"] or not data["runs"][0]["results"]:
+                print(colored("  No issues found.", "green"))
+                return 0
+
+            results = data["runs"][0]["results"]
+            total_issues = len(results)
+            color = "red" if total_issues > 0 else "green"
+            print(
+                colored(f"  Total issues found: {total_issues}", color, attrs=["bold"])
+            )
+
+            for result in results:
+                rule_id = result["ruleId"]
+                message = result["message"]["text"]
+                location = result["locations"][0]["physicalLocation"]
+                file_path = location["artifactLocation"]["uri"]
+                line = location.get("region", {}).get("startLine", "?")
+                print(
+                    f"\n  - {colored('Issue:', 'blue')} "
+                    f"{colored(rule_id, 'blue', attrs=['bold'])}"
+                )
+                print(f"    {colored('File:', 'blue')} {file_path}:{line}")
+                print(f"    {colored('Description:', 'blue')} {message}")
+
+            return total_issues
+
+        except FileNotFoundError:
+            print(
+                colored(f"  Error: Results file not found at: {results_path}", "red"),
+                file=sys.stderr,
+            )
+            return 0
+        except Exception as e:
+            print(
+                colored(f"  An error occurred while printing the summary: {e}", "red"),
+                file=sys.stderr,
+            )
+            return 0
+
+    def _run_command(self, command, cwd=None):
+        """Runs a command and exits on failure, streaming output in real-time."""
+        try:
+            subprocess.run(
+                command,
+                check=True,
+                cwd=cwd,
+            )
+        except subprocess.CalledProcessError as e:
+            print(f"Command failed with exit code {e.returncode}", file=sys.stderr)
```

Comment:
Here as well, the nit about error log following the common convention in this script.